applicationset/controllers/applicationset_controller.go

```diff
@@ -134,6 +138,45 @@ func (r *ApplicationSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{RequeueAfter: ReconcileRequeueOnValidationError}, nil
 	}
 
+	applications, err := r.getCurrentApplications(ctx, applicationSetInfo)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get current applications for application set: %w", err)
+	}
```

Comment:
Can we move this inside the `if`? I see that `appSyncMap` and `appMap` need to be at the function scope, but I don't think this does.